cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,635 @@
+from cfme.modeling.base import BaseEntity
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.appliance.implementations.ui import navigator
+from cfme.utils.log import logger
+from cfme.utils.version import Version
+from cfme.utils.version import VersionPicker
+from cfme.utils.wait import wait_for
+from widgetastic_manageiq import HiddenFileInput
+from widgetastic_manageiq import MigrationDashboardStatusCard
+from widgetastic_manageiq import MigrationDropdown
+from widgetastic_manageiq import MigrationPlanRequestDetailsList
+from widgetastic_manageiq import MigrationPlansList
+from widgetastic_manageiq import MigrationProgressBar
+from widgetastic_manageiq import RadioGroup
+from widgetastic_manageiq import Table
+from widgetastic_manageiq import V2VPaginatorPane
+
+
+class MigrationView(BaseLoggedInPage):
+
+    @property
+    def in_explorer(self):
+        nav_menu = VersionPicker(
+            {
+                Version.lowest(): ["Compute", "Migration"],
+                "5.10": ["Compute", "Migration", "Migration Plans"],
+            }
+        ).pick()
+        return self.logged_in_as_current_user and self.navigation.currently_selected == nav_menu
+
+    @property
+    def is_displayed(self):
+        return self.in_explorer
+
+
+class DashboardCards(View):
+    not_started_plans = MigrationDashboardStatusCard("Not Started")
+    in_progress_plans = MigrationDashboardStatusCard("In Progress")
+    completed_plans = MigrationDashboardStatusCard("Complete")
+    archived_plans = VersionPicker(
+        {
+            Version.lowest(): MigrationDashboardStatusCard("Infrastructure Mappings"),
+            "5.10": MigrationDashboardStatusCard("Archived"),
+        }
+    )
+
+    def read(self):
+        return [c.read().get("name") for c in self.sub_widgets if c.is_active]
+
+
+class MigrationPlanView(MigrationView):
+    dashboard_cards = View.nested(DashboardCards)
+    paginator_view = View.include(V2VPaginatorPane, use_parent=True)
+    title = Text('.//div[contains(@class, "pull-left")]//h3')
+    create_migration_plan = Text(locator='(//a|//button)[text()="Create Migration Plan"]')
+    configure_providers = Text(locator='//a[text()="Configure Providers"]')
+    sort_type_dropdown = SelectorDropdown("id", "sortTypeMenu")
+    sort_direction = Text(locator=".//span[contains(@class,'sort-direction')]")
+    progress_card = MigrationProgressBar()
+    search_box = TextInput(locator=".//div[contains(@class,'input-group')]/input")
+    clear_filters = Text(".//a[text()='Clear All Filters']")
+    # 5.9 field
+    migr_dropdown = MigrationDropdown(text="Not Started Plans")
+
+    @property
+    def in_migration_plan(self):
+        return (
+            self.in_explorer and self.create_migration_plan.is_displayed and
+            len(self.browser.elements('.//div[contains(@class,"card-pf")]')) > 0
```

Comment:
Q: Are these cards the same as the dashboard cards? If so, why not have a property in the nested view that returns the total number of cards? That way you can avoid the call to `self.browser.elements()`

Reply:
No this is different than dashboard cards. This is indicating progress card that appears on starting migration